### ISSUE TITLE
My RASPI Meshcentral2 no longer does updates and my Letsencrypt certificate expires in 11 days

### ISSUE THREAD

**Issue by CSW2000** [Ylianst/MeshCentral#6575]
Hello
My RASPI Meshcentral2 no longer does updates and my Letsencrypt certificate expires in 11 days, Is this due to the update of the MC since Letsencrypt needs the ACME 2? is my productive remote maintenance server. It worked the last few years and now such a stress.
My System  : Raspberry 4/ 8 GB / SSD am USB
Debian 12 RASPIAN / Node Version 22.11.0  / NPM Version 10.9.0 7 /  MariaDB Version 10.11.3 /  PHP 8.2.7
System is up-to-date, I don't know what to do right now I hope someone has a solution

![grafik](https://github.com/user-attachments/assets/4432aa85-458e-408b-a9dc-fb0f8debe7c1)

Let's Debug Result:  OK

Hier noch meine config.json
```
{
  "$schema": "http://info.meshcentral.com/downloads/meshcentral-config-schema.json",
  "__comment1__": "This is a simple configuration file, all values and sections that start with underscore (_) are ignored. Edit a section and remove the _ in front of the name. Refer to the user's guide for details.",
  "__comment2__": "See node_modules/meshcentral/sample-config-advanced.json for a more advanced example.",
  "settings": {
    "cert": "XXXXXXXX.de",
    "WANonly": true,
    "_LANonly": true,
    "_FastCert": true, 
    "_sessionKey": "KXKXKXKXKXKXKXKXKKXKXKXKXKKXKXKKX",
    "port": 443,
    "aliasPort": 443,
    "redirPort": 80,
    "redirAliasPort": 80,
    "mariadb" : { "host": "127.0.0.1", "user": "XXXXXXUSERXXXXX", "password": "SAGICHNICHT", "secure" : true },
    "compression": true,
    "_wscompression": true,
    "agentwscompression": true,
    "allowframing": true,
    "clickonce": true, 
    "webrtc": true,
    "selfupdate": true
  },
  "domains": {
    "": {
      "title": "XXXXXXXXXX.de",
      "_certUrl": "https://XXXXXXXXXX.de:443/",
      "agentFileInfo": {
      "icon": "MeshServer1.ico"
    },
   "assistantCustomization": {
      "title": "XXXXXXXXXX",
      "image": "XXX_msc.png",
      "fileName": "XXXXXXXXXX_Assistent",
      "allowHighQualityDesktop": false
     },
   "consentMessages": {
        "Title": "XXXXXXXXXX - Service TEL : 0000000000000",
        "Desktop": "{0} ({1}) hat einen Remote-Desktop-Zugriff angefordert. Die Fernwartung von XXXXXXXXXXr ist kostenpflichtig und wird mit 60 Euro a Stunde berechnet. Zugriff gewÃ¤hren? ",
        "Terminal": "{0} ({1}) hat einen Remote-Terminal-Zugriff angfordert. Die Fernwartung von XXXXXXXXXX Service ist kostenpflichtig  und wird mit 52 Euro a Stunde berechnet Zugriff gewÃ¤hren? ",
        "Files": "{0} ({1}) hat einen Remote-Dateizugriff angefordert. Die von XXXXXXXXXX ist kostenpflichtig und wird mit 60 Euro a Stunde berechnet. Zugriff gewÃ¤hren?"
      },
  "notificationMessages": {
        "title": "Nachricht",
        "desktop": "{0} Eine Remote Desktop Session wurde gestartet.",
        "terminal": "{0} Eine Terminal Session wurde gestartet.",
        "files": "{0} Eine DatenÃ¼bertragung ist Aktiv."
      },
  "agentCustomization": {
        "displayName": "XXXXXXXXXX",
        "description": "Die Fernwartung von XXXXXXXXXX ist kostenpflichtig und wird mit 60 Euro a Stunde berechnet - Service Tel: 000000000000",
        "companyName": "XXXXXXXXXX Service",
        "serviceName": "XXXXXXXXXX.de",
        "image": "XXXX_msc.png",
        "installText": "Die Fernwartung von XXXXXXXXXX ist kostenpflichtig und wird mit 60 Euro a Stunde berechnet - Service Tel : 000000000000",
        "fileName": "Fernwartung_Agent",
	"foregroundColor": "255,0,0",
	"backgroundColor": "255,255,0"
     },
      "_title2": "Server",
      "_minify": true,
      "siteStyle": 2,
      "titlepicture": "XXXX.png",
      "loginPicture": "login.png",
      "meshMessengerTitle": "XXXX-Chat",
      "MeshMessengerPicture": "messenger.png",
      "footer": "XXXXXXXXXXX",
      "newAccounts": false,
      "userNameIsEmail": false
    }
  },
  "letsencrypt": {
    "__comment__": "Requires NodeJS 8.x or better, Go to https://letsdebug.net/ first before trying Let's Encrypt.",
    "email": "XXXXXXXX@mail.com",
    "names": "XXXXXXXXXX.de",
    "production": true
  }
}

```

**Comment by CSW2000** [comment]
Thank You for your Help

LE : 

`{
    "configOk": true,
    "leDomains": [
        "fernwartungnm.de"
    ],
    "challenges": {},
    "production": true,
    "webServer": true,
    "certPath": "/home/pi/meshcentral-data/letsencrypt-certs",
    "skipChallengeVerification": false,
    "cert": "Present",
    "daysLeft": 11
}`

LEEVENTS 👍 
`2.12.2024 08:58:50 - Getting certs from local store (Production)
2.12.2024 08:58:50 - Reading certificate files
2.12.2024 08:58:50 - Setting LE cert for default domain.
2.12.2024 08:58:58 - Certificate has 11 day(s) left.
2.12.2024 08:58:58 - Asking for new certificate because of expire time.
2.12.2024 08:58:58 - Generating private key...
2.12.2024 08:59:02 - Setting up ACME client...
2.12.2024 08:59:02 - Creating certificate request...
2.12.2024 08:59:04 - Requesting certificate from Let's Encrypt...
2.12.2024 09:02:43 - Failed to obtain certificate: Request failed with status code 404`

LECHECK : 
`Request:Expire`

**Comment by si458** [comment]
ok so `Request failed with status code 404` isnt an error we specify,
its an error that comes directly from the acme client module
so first thing i would check is port 80 is open wide the public and port 80 is diverted to your meshcentral server if you are self-hosting and do portforwarding etc

**Comment by CSW2000** [comment]
Mhhh  .. . Let's Debug Result: OK
I host the self in my company via the DYNDNS service from my hoster.
And the LE certificate always worked without problems for 2 years or so.
I haven't changed anything on the router either.
Here are the router approvals

![mcfreigabe](https://github.com/user-attachments/assets/22166cf4-0c4a-40b9-99ff-680b73ff6265)

**Comment by si458** [comment]
one thing you could try is set `skipChallengeVerification: true` inside `letsencrypt` in your config.json
restart meshcentral and see if it renews
(dont force it, just wait like 2mins and then see if it renewed or the `levents` change)

only other thing you could do would be to enable the debug of acme-client and watch the logs in the console
so stop meshcentral service then run `DEBUG=acme-client node node_modules/meshcentral` from your root folder
(`/home/pi/` i believe in your case)

**Comment by CSW2000** [comment]
ich have remove teh Port 8181 and 8080  from Not longer presents Webservers 
and now the lecheck result 👍 
`> leevents
2.12.2024 16:28:56 - Getting certs from local store (Production)
2.12.2024 16:28:56 - Reading certificate files
2.12.2024 16:28:56 - Setting LE cert for default domain.
2.12.2024 16:29:04 - Certificate has 11 day(s) left.
2.12.2024 16:29:04 - Asking for new certificate because of expire time.
2.12.2024 16:29:04 - Generating private key...
2.12.2024 16:29:06 - Setting up ACME client...
2.12.2024 16:29:06 - Creating certificate request...
2.12.2024 16:29:08 - Requesting certificate from Let's Encrypt...
2.12.2024 16:29:36 - Request for certificate is in process.
2.12.2024 16:29:52 - Request for certificate is in process.
2.12.2024 16:32:16 - Request for certificate is in process.
`
but the ACME client are wrong  i think ich must update MC higher 1.29 but don not work.
?!?!?!?

**Comment by CSW2000** [comment]
ok after change config.json skipChallengeVerification: true the LECHECK and the ACME DEBUG LOG
![lecheck](https://github.com/user-attachments/assets/636c2846-20b1-477a-b5a2-30320169f97b)
![debugacmet](https://github.com/user-attachments/assets/67c9eb96-33a0-4cc7-898b-756ad231ba16)

**Comment by si458** [comment]
so the first picture shows that basically LetsEncrypt couldnt get the challenge from your meshcentral server for some reason?

as for the 2nd picture, this hasnt run meshcentral correctly as it running on port 1025,

you might need to do as it explains `sudo setcap 'cap_net_bind_service=+ep'...`
this allows node to run with less than port 1025
also remember to STOP meshcentral before hand `systemctl stop meshcentral`
and also maybe use `sudo`

**Comment by CSW2000** [comment]
OK, thanks, the problem was solved.
I still had an Apache server running on port 80 in the background.
I think it crept in when I upgraded from Debian11 to 12, there should actually be an NGINX running on port 8181...
The permissions in my router made me look there. Now the certificate has been renewed, thanks so far.

But the update from 1.29 to the current version still doesn't work.
![LE](https://github.com/user-attachments/assets/ab5cb774-809f-4e3d-880f-df842ca19e94)
